lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1684,65 +1684,69 @@ def plot_surface(self, X, Y, Z, *args, **kwargs):
         if shade and cmap is not None and fcolors is not None:
             fcolors = self._shade_colors_lightsource(Z, cmap, lightsource)
 
+        def boundary_edge(arr):
+            """
+            Get the boundary elements of the rectangle ``arr``,
+
+                [arr[0,0] ... arr[0,-1] ... arr[-1, -1] ... arr[-1,0] ...]
+
+            If arr.shape is (M, N), the result is of length 2(N+M-2)
+            """
+            # note we use Python's half-open ranges to avoid repeating
+            # the corners
+            forward = np.s_[0:-1]      # [0 ... -1)
+            backward = np.s_[-1:0:-1]  # [-1 ... 0)
+            return np.concatenate((
+                arr[0, forward],
+                arr[forward, -1],
+                arr[-1, backward],
+                arr[backward, 0],
+            ))
+
+        # evenly spaced, and including both endpoints
+        row_inds = list(xrange(0, rows-1, rstride)) + [rows-1]
+        col_inds = list(xrange(0, cols-1, cstride)) + [cols-1]
```

Comment:
this isn't py3k compatible. `xrange` doesn't exist, right?

Reply:
Maybe `np.array_split()` would be cleaner? Don't know when it was added to numpy, but I do see it in v1.12.

Reply:
`array_split` produces non-overlapping chunks, but an interesting idea.

I assume that there was `xrange = range` or something, but I guess not. I'll switch to `range`

Reply:
It looks like I this came in after rebasing on #10525. Does matplotlib no longer support python 2?

Reply:
Not on `master`, it doesn't.

Reply:
Am I right to target this against `master`?

Reply:
Yes, we can always backport if necessary (I don't think it is).